prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchema.kt

```diff
@@ -58,45 +55,20 @@ data class ConfigSchema(
 
     /**
      * Validate the top level schema if [isChildSchema] is false, or a child schema if [isChildSchema] is true.
+     * [validationErrors] can optionally be specified to start with a list of errors, but defaults to an empty list.
      * @return a list of validation errors, or an empty list if no errors
      */
-    internal fun validate(isChildSchema: Boolean = false): List<String> {
-        val validationErrors = mutableListOf<String>()
-
+    internal open fun validate(
+        isChildSchema: Boolean = false,
+        validationErrors: MutableList<String> = mutableListOf()
+    ): List<String> {
         /**
          * Add an error [msg] to the list of errors.
          */
         fun addError(msg: String) {
             validationErrors.add("Schema $name: $msg")
         }
 
```

Comment:
errors is mutable. If you extract this out it may make it harder as this manipulates a local list

Reply:
I don't think errors is mutable, since you're not able to add elements to it. If it was mutable it would make some of this logic simpler. There wouldn't be a need for a local list.

Reply:
Since it uses a local variable `validationErrors`, it has to be defined inside the function as it currently exists. I'll see if there's a way that makes sense to make validationErrors a private class property instead, related to your other comment. If that works out alright, then I can pull the local function out into the parent class too.

Reply:
Adjusted this in latest commit